### PR TITLE
Issue #29 - Add tox support

### LINKED ISSUE #29: As a developer I want to add default tox configuration
## 💪 Motivation
`tox` provides test environment management and automation capabilities including management of environments for multiple python / dependency version testing (if we find that useful in the future). It provides a single place for project management commands and can unify our local development and CI testing configuration.

### PR DESCRIPTION
**Pull request by MJJoyce**
Add a `tox` support. Add testenvs for running our test suite against
python 3.9, building our docs, and executing our "pre-commit linting suite".

Add a "dev" testenv for creating a development environment via `tox` if
users are so inclined.

Rework the way we're handling checks in pre-commit.
We call our pre-commit "linters" from tox so we need to:

- Move as many checks as possible over to pull specific revisions of the
repositories. The caching in `pre-commit` is much nicer and saves us the
time of `tox` building a new environment, etc. Note, these specific revs
should be updated regularly via `pre-commit autoupdate`.

- Add filtering to keep some of checks from running across the
repository.

- Re-baseline our repository with a run of `pre-commit run
--all-files`. This is the same thing `tox` is going to use to call our
steps.

Resolve #29

### CODE REVIEW

**Comment by MJJoyce** [comment]
Note, the README is getting a bit all over the place with all the changes to tooling and the like. I'll see what I can do to clean that up in a follow on issue.

File: README.md [hunk 1]
```diff
@@ -51,10 +51,14 @@ To isolate and be able to re-produce the environment for this package, you shoul
...
 Then exclusively use `venv/bin/python`, `venv/bin/pip`, etc. (It is no longer recommended to use `venv/bin/activate`.)
 
-Dependencies for development are stored in requirements.txt; they are installed into the virtual environment as follows:
+If you have `tox` installed and would like it to create your environment and install dependencies for you run:
+
+    tox --devenv <name you'd like for env> -e dev
+
+Dependencies for development are specified as the `dev` `extras_requier` in `setup.cfg`; they are installed into the virtual environment as follows:
+
+    pip install --editable .[dev]
```
Comment (nutjob4life) [inline README.md:60]: Maybe we should put some quotes here so users' shells don't interpolate the `[…]` brackets?

```console
pip install --editable '.[dev]'
```

**Review by nutjob4life — APPROVED** [review]
NOICE.

File: README.md (hunk 1 above)
Comment (MJJoyce) [reply README.md:60]: +1, I'll update. Looks like I typoed `extras_require` as `extras_requier` as well so I'l get that in there as well.

**Comment by jordanpadams** [comment]
@MJJoyce @nutjob4life per the updates, do we think it makes sense to move some of the library recommendations, best practices, etc. to a wiki / Github page at https://nasa-pds.github.io/ and just link to it? not sure what the best practice is for that either :-)

**Comment by MJJoyce** [comment]
@jordanpadams, I'd say if we're expecting people to use this template repo as the go-to for anything Python it's probably best to keep it there. Moving it to a wiki also runs the risk of stuff getting out of sync with what's in here, etc.

**Pull request merged**